src/test/java/com/dimsirka/animalservice/integration/AnimalControllerTest.java

```diff
@@ -187,4 +189,66 @@ void getAllByIdTest_successFlow() {
                 response.getBody().toString()
         );
     }
+
+    @Test
+    void getAnimalsByStatusTest_successFlow() {
+        Animal animal1 = Animal.builder()
+                .animalStatus(AnimalStatus.FREE)
+                .animalType(AnimalType.CAT)
+                .name("Drago1")
+                .description("Drago is a friendly and easy-going pet. It needs to find a real family!").build();
+
+        Animal animal2 = Animal.builder()
+                .animalStatus(AnimalStatus.FREE)
+                .animalType(AnimalType.CAT)
+                .name("Drago2")
+                .description("Drago is a friendly and easy-going pet. It needs to find a real family!").build();
+        animalRepository.save(animal1);
+        animalRepository.save(animal2);
```

Comment:
It would be better to make this part of a separate method like a source of testing objects. Look at code below:
```
private static AnimalDto getAnimalDtoFixture(){
        return AnimalDto.builder()
                .animalStatus(AnimalStatus.FREE)
                .animalType(AnimalType.CAT)
                .name("Drago-1")
                .description("Drago is a friendly and easy-going pet. It needs to find a real family!").build();
    }
```
And use this method wherever you need it!
In general a good job.